test/test_functional.py

```diff
@@ -8,9 +7,13 @@
 
 
 class TestFunctional(unittest.TestCase):
-    data_sizes = (2, 20)
+    data_sizes = [(2, 20), (3, 15)]
     number_of_trials = 100
 
+    def setUp(self):
+        # we want to make sure that the random values are reproducible
+        torch.manual_seed(0)
```

Comment:
This isn't always going to work. It requires a deterministic execution of calls to the random generator to be reproducible. As soon as you change that order, it'll generate different data at its call-sites.

A random number generator could be very simple like adding the last two numbers, adding some constant and then modding it by some fixed number

Reply:
*some fixed prime number